### PR TITLE
Create mock test case for the LoadNetwork API in the AUTO plugin

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - Create mock test for LoadNetwork validation in the AUTO plugin.

### Tickets:
 - https://jira.devtools.intel.com/browse/CVS-75771

### CODE REVIEW

File: src/tests/unit/auto/load_network_test.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+using ::testing::InvokeWithoutArgs;
+using ::testing::MatcherCast;
+using ::testing::Matches;
+using ::testing::Property;
+using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::StrEq;
+using ::testing::Throw;
+using namespace MockMultiDevice;
+
```
Comment (songbell) [inline src/tests/unit/auto/load_network_test.cpp:38]: can we remove the un-needed using?

File: src/tests/unit/auto/load_network_test.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        reinterpret_cast<void (*)(IInferencePlugin*)>(ov::util::get_symbol(sharedObjectLoader, "InjectProxyEngine")));
+
+    // prepare mockicore and cnnNetwork for loading
+    Core ie;
+    auto* origin_plugin = new MockMultiDeviceInferencePlugin();
+    // replace core with mock Icore
+    injectProxyEngine(origin_plugin);
+    //ie.RegisterPlugin(std::string("mock_engine") + IE_BUILD_POSTFIX, "AUTO");
+
+    EXPECT_CALL(*origin_plugin, LoadNetworkImpl(_, _, _, _)).Times(AnyNumber());
```
Comment (songbell) [inline src/tests/unit/auto/load_network_test.cpp:53]: may be Times should be 1?

**Comment by ilya-lavrenov** [comment]
is it still valid?

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Pull request closed without merging**